nntrainer/layers/multi_head_attention_layer.cpp

```diff
@@ -77,10 +77,415 @@ enum AttentionParams {
   transposed_attention_output,
 };
 
-void MultiHeadAttentionLayer::finalizeCommon(InitLayerContext &context) {}
+void MultiHeadAttentionLayer::finalizeCommon(InitLayerContext &context) {
+  if (context.getNumInputs() < 2 || context.getNumInputs() > 4) {
+    throw std::runtime_error("Multi head Attention layer needs 2-4 inputs.");
+  }
+
+  NNTR_THROW_IF(
+    std::get<props::ReferenceFramework>(multi_head_attention_props).empty(),
+    std::invalid_argument)
+    << "reference_framework property is not provided for layer "
+    << context.getName();
+  const Framework ref_framework =
```

Comment:
I think defalut value of this property needs to be set.

Reply:
fixed